### ISSUE TITLE
Error objects should be more consistent

### ISSUE THREAD

**Issue by ambarth** [micky2be/superlogin-client#77]
Superlogin-client api returns very different error objects depending on what goes wrong. There are at least three different cases:

1. No network connection: If superlogin-client can not even connect to the superlogin server there will be an error object `err` with a `message` property with value 'Network Error' and `err.response` and `err.status` will be undefined (`err.response` at least exists as key). I think this is due to axios.

2. superlogin-client detects some constraint violation on the client side (for example the username field or password is missing) and rejects the api call with an error object with a single `error` property that has a string value explaining what is the problem, for example here: https://github.com/micky2be/superlogin-client/blob/56fd47730d0bc410d933aca2c722462e99511993/src/index.js#L336

3. superlogin-client successfully connects to the superlogin server, but it returns an error code (for example the credentials do not match etc.). In this case the error object is simply the response body sent by the server. Unless there is some other server side error it will have an `error` and a `message` property both of which have a string value sent by superlogin. However, the error code sent by the server gets lost here: 
https://github.com/micky2be/superlogin-client/blob/56fd47730d0bc410d933aca2c722462e99511993/src/index.js#L43

In short handling superlogin-client error handling would look something like this: 
```javascript
if (!err.response && err.message === 'Network Error') {
    // network connection failed
} else if (err.error && typeof err.status === 'undefined') {
  // constraint violation, err.error is a string explaining the problem
} else {
  // err is the response body sent by the server, but it might look different depending on what happened,
  // it could be the superlogin response or anything else you send on the server
  // since we do not even have the status code its hard to tell what the problem is
}
```

A first step to improve this would be to let attach the status code to the error object in `parseError`.
We could use status 0 for the case where no connection could be established.
